index.js

```diff
@@ -14,9 +12,27 @@ module.exports = {
       this.child.call('getHAR', done)
     })
 
+    Nightmare.action('resetHAR', (name, options, parent, win, renderer, done) => {
+      parent.respondTo('resetHAR', (done) => {
+        renderer.on('har-reset', (event) => {
+          done(null)
+        })
+        win.webContents.executeJavaScript('new Image().src = "chrome://version?reset-har"')
+      })
+      done()
+    }, function (done) {
+      this.child.call('resetHAR', done)
+    })
+
     Nightmare.action('waitForDevtools', (name, options, parent, win, renderer, done) => {
       parent.respondTo('waitForDevtools', (done) => {
-        win.webContents.on('devtools-opened', () => done(null))
+        win.webContents.on('devtools-opened', () => {
+          let path = require('path')
+          let BrowserWindow = require('electron').BrowserWindow
+          BrowserWindow.removeDevToolsExtension('devtools-har-extension')
+          BrowserWindow.addDevToolsExtension(path.dirname(require.resolve('nightmare-har-plugin/devtools-har-extension')))
```

Comment:
When this code is executed in `preload.js` the extension may miss the `load` events since it removes the extension and install again. So the time to this operation be done can be greater than the document loaded or page load events, specially on the use case where the program navigate to multiple pages before request the har data.